src/test/java/org/takes/tk/TkProxyTest.java

```diff
@@ -34,34 +34,69 @@
 import org.takes.http.FtRemote;
 import org.takes.rq.RqFake;
 import org.takes.rq.RqHref;
+import org.takes.rq.RqMethod;
 import org.takes.rs.RsPrint;
 import org.takes.rs.RsText;
 
 /**
  * Test case for {@link TkProxy}.
  * @author Dragan Bozanovic (bozanovicdr@gmail.com)
+ * @author sebing
  * @version $Id$
  * @since 0.25
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
- * @todo #377:30min/DEV We need more tests for TkProxy.
- *  The tests should verify the different HTTP methods (GET, POST, etc),
- *  as well as the different combinations of request/response headers.
- */
+  */
 public final class TkProxyTest {
 
     /**
-     * TkProxy can work.
-     * @throws Exception If some problem inside
+     * TkProxy can just works on POST.
+     * @throws Exception If some problem inside.
+     */
+    @Test
+    public void justWorksOnPost()throws Exception {
+        this.justWorks(RqMethod.POST);
+    }
+
+    /**
+     *TkProxy can just works on GET.
+     * @throws Exception If some problem inside.
+     */
+    @Test
+    public void justWorksOnGet()throws Exception {
+        this.justWorks(RqMethod.GET);
+    }
+
+    /**
+     * TkProxy can correctly maps path string on GET.
+     * @throws Exception If some problem inside.
      */
     @Test
-    public void justWorks() throws Exception {
+    public void correctlyMapsPathStringOnGet() throws Exception {
+        this.correctlyMapsPathString(RqMethod.GET);
+    }
+
+    /**
+     *TkProxy can correctly maps path string on POST.
+     * @throws Exception If some problem inside.
+     */
+    @Test
+    public void correctlyMapsPathStringOnPost() throws Exception {
+        this.correctlyMapsPathString(RqMethod.POST);
+    }
+
+    /**
+     * A private method to do test with multiple httpMethods.
+     * @param httpmethod HTTP methods (GET, POST, etc),
+     * @throws Exception If some problem inside.
+     */
+    private void justWorks(final String httpmethod) throws Exception {
```

Comment:
@sebing why not just `method`? see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html